core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureChecker.scala

```diff
@@ -74,73 +118,107 @@ trait TransactionSignatureChecker {
+          !HashType.isDefinedHashtypeSignature(signature)
+        ) {
+          SignatureValidationErrorHashType
+        } else if (pubKeyEncodedCorrectly.isDefined) {
+          val err = pubKeyEncodedCorrectly.get
+          val result =
+            if (err == ScriptErrorWitnessPubKeyType)
+              SignatureValidationErrorWitnessPubKeyType
+            else SignatureValidationErrorPubKeyEncoding
+          result
+        } else {
+          val sigsRemovedScript = BitcoinScriptUtil.calculateScriptForChecking(
+            txSignatureComponent,
+            signature,
+            script)
+          val hashTypeByte =
+            if (signature.bytes.nonEmpty) signature.bytes.last else 0x00.toByte
+          val hashType = HashType(
+            ByteVector(0.toByte, 0.toByte, 0.toByte, hashTypeByte))
+          val spk = ScriptPubKey.fromAsm(sigsRemovedScript)
+          val hashForSignature = txSignatureComponent match {
+            case w: WitnessTxSigComponent =>
+              TransactionSignatureSerializer.hashForSignature(
+                w,
+                hashType,
+                TaprootSerializationOptions.empty)
+            case b: BaseTxSigComponent =>
+              val sigsRemovedTxSigComponent = BaseTxSigComponent(
+                b.transaction,
+                b.inputIndex,
+                TransactionOutput(b.output.value, spk),
+                b.flags)
+              TransactionSignatureSerializer.hashForSignature(
+                sigsRemovedTxSigComponent,
+                hashType,
+                TaprootSerializationOptions.empty)
+            case r: WitnessTxSigComponentRebuilt =>
+              val sigsRemovedTxSigComponent = WitnessTxSigComponentRebuilt(
+                wtx = r.transaction,
+                inputIndex = r.inputIndex,
+                output = TransactionOutput(r.output.value, spk),
+                witScriptPubKey = r.witnessScriptPubKey,
+                flags = r.flags)
+              TransactionSignatureSerializer.hashForSignature(
+                sigsRemovedTxSigComponent,
+                hashType,
+                TaprootSerializationOptions.empty)
+          }
+
+          val sigWithoutHashType = stripHashType(signature)
+          val isValid = pubKey.verify(hashForSignature, sigWithoutHashType)
+          if (isValid) SignatureValidationSuccess
+          else
+            nullFailCheck(Seq(signature),
+                          SignatureValidationErrorIncorrectSignatures,
+                          flags)
+        }
+    }
+  }
 
-      val sigWithoutHashType = stripHashType(signature)
-      val isValid = pubKey.verify(hashForSignature, sigWithoutHashType)
-      if (isValid) SignatureValidationSuccess
-      else
-        nullFailCheck(Seq(signature),
-                      SignatureValidationErrorIncorrectSignatures,
-                      flags)
+  def checkSigTapscript(
+      txSignatureComponent: TxSigComponent,
+      pubKey: XOnlyPubKey,
```

Comment:
I think this is supposed to be a `SchnorrPublicKey`

Reply:
Its `XOnlyPubKey` in core

https://github.com/bitcoin/bitcoin/blob/34869114a72435b9d8364385abaefed70d703fa8/src/script/interpreter.cpp#L1684

https://github.com/bitcoin/bitcoin/blob/master/src/script/interpreter.cpp#L1093-L1096

Reply:
They parse it as XOnly on the first link

```c++
XOnlyPubKey pubkey{pubkey_in};
```

Reply:
In general XOnly keys on which you only call .schnorrPublicKey and never explicitly pass a parity should be SchnorrPublicKey (or SchnorrNonce in the rare case that it is an R value and not a pubkey)

Reply:
Yup, sounds good to me